rats-rs/src/transport/tls/mod.rs

```diff
@@ -0,0 +1,168 @@
+mod client;
+mod server;
+
+use super::{Error, ErrorKind, Result};
+use crate::{
+    cert::{
+        dice::extensions::{OID_TCG_DICE_ENDORSEMENT_MANIFEST, OID_TCG_DICE_TAGGED_EVIDENCE},
+        verify::{
+            verify_cert_der, CertVerifier, VerifiyPolicy, VerifiyPolicy::Contains,
+            VerifyPolicyOutput,
+        },
+    },
+    tee::claims::Claims,
+};
+use bitflags::bitflags;
+pub use client::{Client, TlsClientBuilder};
+use lazy_static::lazy_static;
+use libc::c_int;
+use log::{debug, error};
+use openssl_sys::*;
+use pkcs8::ObjectIdentifier;
+pub use server::{Server, TlsServerBuilder};
+use std::{
+    cell::Cell,
+    net::TcpStream,
+    os::fd::AsRawFd,
+    ptr, slice,
+    sync::{Arc, Mutex, Once},
+};
+
+lazy_static! {
+    static ref OPENSSL_EX_DATA_IDX: Arc<Mutex<Cell<i32>>> = unsafe {
+        Arc::new(Mutex::new(Cell::new(CRYPTO_get_ex_new_index(
+            4,
+            0,
+            ptr::null_mut(),
+            None,
+            None,
+            None,
+        ))))
+    };
+}
```

Comment:
Can be replaced with:

```suggestion
lazy_static! {
    static ref OPENSSL_EX_DATA_IDX: i32 =
        unsafe { CRYPTO_get_ex_new_index(4, 0, ptr::null_mut(), None, None, None,) };
}
```